aasig_dev_platform/cuttlefish.py

```diff
@@ -0,0 +1,130 @@
+'''
+Created on Jan 21, 2022
+
+@author: vladyslav_goncharuk
+'''
+
+from paf.paf_impl import Task, CommunicationMode
+from asyncio.log import logger
+
+class CuttlefishDeploymentTask(Task):
+
+    def __init__(self):
+        super().__init__()
+
+        self.CUTTLEFISH_PATH = "${ROOT}/${ANDROID_DEPLOYMENT_DIR}/${SOURCE_DIR}/${CUTTLEFISH_SUB_DIR}"
+
+class cuttlefish_install_docker(CuttlefishDeploymentTask):
+
+    def __init__(self):
+        super().__init__()
+        self.set_name(cuttlefish_install_docker.__name__)
+
+    def execute(self):
+        self.subprocess_must_succeed("sudo -S apt-get update; "
+            "sudo -S apt-get install -y ca-certificates curl gnupg lsb-release; ")
```

Comment:
This assumes a DPKG based distribution (debian/ubuntu) and fails on other distros. Is it the only supported distro for cuttlefish build?  
README could document this at least.  

I think a docker environment that sets up the build environment would be useful.

Generally I am not a fan to have a python script execute a subprocess with "sudo install" and many other system-modifying actions.  It seems to me this should at least be extracted out of the build steps and become a requirement on the build host instead.  You can still provide a script to make it convenient to install all dependencies, but not hidden within the build steps?  After all, it only needs to run once, not every build.

Reply:
Hi @gunnarx,

*This assumes a DPKG based distribution (debian/ubuntu) and fails on other distros. Is it the only supported distro for cuttlefish build?*

The current status of this PR is that it was not considering making the automation "Linux-distro agnostic". We did not consider it as a requirement.

**I'll mention in the README, that currently, the approach is distribution specific.**

=> Still, I totally agree that it should be only a temporary solution.

----

*I think a docker environment that sets up the build environment would be useful.*

As it has now become an open point, we need to figure out which way is the best. I would also vote for preparing build prerequisites inside of the Docker image. But, better to first have this PR in, in order to be sure in the used approach, and minimize the chance of throwing additional efforts in the trash bin.

----

*Generally I am not a fan to have a python script execute a subprocess with "sudo install" and many other system-modifying actions. It seems to me this should at least be extracted out of the build steps and become a requirement on the build host instead. You can still provide a script to make it convenient to install all dependencies, but not hidden within the build steps?*

Well, from my POV Python is better structured than the raw bash code. Also, it is object-oriented. All that allows having a more maintainable source code. Moreover, the PAF framework encourages developers to implement automation in form of bash one-liners. So, anyone, who is familiar with bash should be able to keep going with the support of this repository.

PAF is quite flexible. We can move the installation of docker out of the steps to run android cuttlefish emulation. We just need to comment out one line here:
https://github.com/COVESA/aasig_dev_platform/blob/3d9a1ba3d6e5d0b26410817d28668eaed19523ed/aasig_dev_platform/scenarios.xml#L94

**Or, I can declare a separate variable, which will explicitly enable that step inside the cuttlefish build with the default "OFF" value. I'll do this adjustment.**

*After all, it only needs to run once, not every build.*

All implemented steps are re-entrable. It will do no harm to make them each time. Still, it might be annoying to enter sudo credentials each time. Thus, I'll turn it off by default.

----

So, 2 action-items here:
- Mention in the README, that currently, the approach is distribution specific
- Declare a separate variable, which will explicitly enable the docker installation inside the cuttlefish build with the default "OFF" value

Please, let me know if from your POV there are other open points here.

Reply:
Hi @gunnarx.

*Mention in the README, that currently, the approach is distribution specific*

Done:

![image](https://user-images.githubusercontent.com/4373469/154676372-6f0e8366-a5f1-47b4-9e1e-3cd9efb14447.png)

![image](https://user-images.githubusercontent.com/4373469/154676682-f25609e1-f5fa-4217-af2c-f9c9cf71fd3e.png)

*Declare a separate variable, which will explicitly enable the docker installation inside the cuttlefish build with the default "OFF" value*

I've done it another way around. Simply ensured, that helper tasks are not executed as part of main scenarios:

![image](https://user-images.githubusercontent.com/4373469/154676878-b74de5c1-5ec9-46f6-a523-09409bf897e9.png)

If one needs, he can execute that specific task in a separate run.